### PR TITLE
Refact/refactor test objects

### PR DESCRIPTION
**Pull request by n-r-b**
### JIRA
No JIRA

### Change description
- Updated TestData.java with test payment data and refactored tests to use this
- Created Payment/SuppressionEntity fixtures for generating data objects and refactored tests to use these
- Refactored `asJsonString` converter method from the controller tests into it's own class
- Removed test json files as they're no longer needed

### Swagger Screenshot
N/A

### Work Checklist

- [x] Tests added where applicable
- [x] Test coverage of new code meets target of 80%

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/model/payment/Payment.java [hunk 1]
```diff
@@ -19,6 +19,16 @@
 
     private List<PaymentItem> items;
 
+    public Payment() {}
```
Comment (akaJSaunders) [inline src/main/java/uk/gov/companieshouse/model/payment/Payment.java:22]: Do we need these empty constructors?

File: src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java [hunk 2]
```diff
@@ -31,18 +33,26 @@
...
 
     @MockBean
     private SuppressionService suppressionService;
 
-    private final ObjectMapper mapper = new ObjectMapper();
+    private JacksonTester<PaymentPatchRequest> json;
+
+    @BeforeEach
+    void setUp(){
+        JacksonTester.initFields(this, objectMapper);
```
Comment (akaJSaunders) [inline src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java:54]: sorry, could you explain what we are setting up here? What is the purpose of this JacksonTester?

File: src/test/java/uk/gov/companieshouse/service/EmailServiceTest.java [hunk 3]
```diff
@@ -25,9 +26,8 @@
 import uk.gov.companieshouse.fixtures.SuppressionFixtures;
 
 @ExtendWith(MockitoExtension.class)
-public class EmailServiceTest {
+class EmailServiceTest {
```
Comment (akaJSaunders) [inline src/test/java/uk/gov/companieshouse/service/EmailServiceTest.java:29]: why are we removing the `public` prefix?

**Review by akaJSaunders — COMMENTED** [review]
These are some very nice changes, good job!  🥇 I have a few comments, but they are mostly for my own understanding

File: src/main/java/uk/gov/companieshouse/model/payment/Payment.java (hunk 1 above)
Comment (n-r-b) [reply src/main/java/uk/gov/companieshouse/model/payment/Payment.java:22]: Having an empty constructor allows us to construct the object without any initial values. The `PaymentService` requires a no-arg constructor as it uses the getter methods to set the data.

Let me update these newly created empty constructors to be consistent with the ones in other models.

File: src/test/java/uk/gov/companieshouse/service/EmailServiceTest.java (hunk 3 above)
Comment (n-r-b) [reply src/test/java/uk/gov/companieshouse/service/EmailServiceTest.java:29]: as identified by the sonar scanner, "JUnit5 test classes and methods should have default package visibility". JUnit4 required everything to be `public`. 
You can find more information about this here: https://rules.sonarsource.com/java/RSPEC-5786

File: src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java (hunk 2 above)
Comment (n-r-b) [reply src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java]: so this just initialises the JacksonTest fields - our test instance and the object mapper we'd like to use. 
However, I've decided to go back to our original implementation as i think using this JacksonTester is overkill for what we are trying to achieve. I've refactored out the duplicate `asJsonString` method into it's own class - please take a look

File: src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java [hunk 4]
```diff
@@ -25,13 +24,14 @@
...
 
+import static uk.gov.companieshouse.JsonConverter.convertObjectToJsonString;
 import static uk.gov.companieshouse.TestData.Suppression.applicationReference;
 import static uk.gov.companieshouse.fixtures.PaymentFixtures.generatePaymentPatchRequest;
 
 @WebMvcTest(PaymentController.class)
 class PaymentControllerTest_PATCH {
 
-    private final String SUPPRESSIONS_PAYMENT_URI = "/suppressions/{suppression-id}/payment";
+    private static final String SUPPRESSIONS_PAYMENT_URI = "/suppressions/{suppression-id}/payment";
```
Comment (akaJSaunders) [inline src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java:34]: why is this static? Surely no other class will be accessing a variable from a controller test

Comment (n-r-b) [reply src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest_PATCH.java:34]: i'm assuming this is a constant, and constants are `static final` fields (see the google style guide: https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names)

**Review by akaJSaunders — APPROVED** [review]

**Pull request merged**